Add newsletter November (#285)
maplibre/maplibre.github.io: content/news/2023-11-30-maplibre-newsletter-november-2023/index.md

---
title: "MapLibre Newsletter November 2023"
date: "2023-11-30"
categories: ["announcements"]

> authors: [wipfli]

Reviewer: ```suggestion
authors: [wipfli, yuri, bart]
```

draft: false
---

# MapLibre Newsletter - November 2023

## MapLibre GL JS

### v4

Versions 3.5.2, 3.6.0, 3.6.1 were released, and we started going into the pre-release mode for MapLibre GL JS v4. The most significant breaking change in version 4 will be the replacement of outdated JavaScript callbacks with Promises in several public APIs.

You can now start testing the version 4 pre-releases at https://github.com/maplibre/maplibre-gl-js/releases. Your feedback and bug reports are more than welcome!

### Globe

Not just one, but actually two proofs of concept for a globe view were shared by two independent developers this month. See https://github.com/maplibre/maplibre/discussions/161.

We are excited to see where this will bring us!

<a href="https://pheonor.github.io/globe-demo/"><img src="globe.jpg" style="max-width: 100%"></a>
<br>

> <i>Globe view proof-of-concept by Vivan Larrieu. Link: https://pheonor.github.io/globe-demo/.</i>

Reviewer: ```suggestion
<i>Globe view proof of concept by Vivan Larrieu. Link: https://pheonor.github.io/globe-demo/.</i>
```

Re-enabling macOS support using the Metal renderer (native macOS, Qt, Node.js) will be one of the last topics to be addressed as part of the Metal project.

### Other Developments

- Tadej Novak has made a lot of progress with the [MapLibre Native Qt repo](https://github.com/maplibre/maplibre-native-qt). In the future, the Qt platform will live in this repository instead of the main repository.

- [Multi-sprites support for MapLibre Native](https://github.com/maplibre/maplibre-native/pull/1858) is expected to land soon. More information about this feature can be found in the [MapLibre Style Spec documentation](https://maplibre.org/maplibre-style-spec/sprite/).

- Two design proposals were accepted this month:
  - [Complex, Animatable, Interactive Annotations Proposal](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2023-11-08-complex-animatable-interactive-annotations.md)
  - [Ambient cache initialization speed improvement](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2023-11-06-ambient-cache-initialization-speed-improvement.md)

## Martin

> The new Martin tile server now has a tool similar to `tilelive-copy` to bulk copy large amounts of tiles from any source supported by the Martin tile server, using the same source configuration system/CLI parameters, but without starting up the whole server in order to generate them.

Reviewer 2: ```suggestion
Martin tile server now has a tool similar to `tilelive-copy` and `tippecanoe` to bulk copy tiles in bulk from any source supported by the Martin tile server, using the same source configuration system/CLI parameters, but without starting up the whole tile server in order to generate them.
```

> Martin tile server now has a tool similar to `tilelive-copy` and `tippecanoe` to copy tiles in bulk from any source supported by the Martin tile server, using the same source configuration system/CLI parameters, but without starting up the whole tile server in order to generate them. Available as a standalone binary with no external dependencies.

Reviewer 2: ```suggestion
Martin tile server has a new tool similar to `tilelive-copy` and `tippecanoe` to bulk-copy tiles from any supported source. The `martin-cp` tool uses the same source configuration system/CLI parameters as Martin server, but without starting up the whole tile server in order to generate them. The tool is available as a standalone binary with no external dependencies.
```

> Use case: generate several areas of the map from PostgreSQL tables/functions or other sources, and even merge multiple tile layers together into a single tileset and store them into an .mbtiles file.

Reviewer 2: ```suggestion
Use case: generate several areas of the map from PostgreSQL tables/functions or other sources, and even merge multiple tile layers together into a single .mbtiles tileset file.
```

> Another new tool improvement: `mbtiles summary` prints now in-depth details about an mbtiles file.

Reviewer 2: ```suggestion
In other news, `mbtiles summary` tool can now print in-depth details about an mbtiles file such as per-zoom statistics.
```
